crates/sui-core/src/authority/authority_test_utils.rs

```diff
@@ -305,6 +305,26 @@ pub async fn certify_shared_obj_transaction_no_execution(
     Ok(certificate)
 }
 
+pub async fn enqueue_all_and_execute_all(
+    authority: &AuthorityState,
+    certificates: Vec<VerifiedCertificate>,
+) -> Result<Vec<(TransactionEffects, Option<ExecutionError>)>, SuiError> {
+    authority
+        .enqueue_certificates_for_execution(
+            certificates.clone(),
+            &authority.epoch_store_for_testing(),
+        )
+        .unwrap();
+
+    let mut output = Vec::new();
+    for cert in certificates {
+        let (result, execution_error_opt) = authority.try_execute_for_test(&cert).await?;
```

Comment:
I am not sure how this test demonstrates that, it actively tries to execute each transaction, so of course they won't get stuck?
Could you explain how tx manager notification works in the following scenario:
Tx1 deletes a shared object, Tx2 uses the deleted shared object.
When Tx1 finishes execution, how does it notify Tx2 that it's ready to execute?

Reply:
I agree that there is a bug here. Normally transaction manager is notified of new objects by `commit_cert_and_notify` - but we no longer write the new objects that the pending txns were expecting, so the pending txns will never be woken.

Reply:
started a conversation in the send to object channel on this